lib/IRGen/GenProto.cpp

```diff
@@ -2880,8 +2946,13 @@ irgen::emitGenericRequirementFromSubstitutions(IRGenFunction &IGF,
   auto proto = requirement.Protocol;
   auto conformance = *subs.lookupConformance(depTy, proto);
   assert(conformance.getRequirement() == proto);
+  auto specialized = conformance;
+  if (!conformance.isConcrete() ||
+      conformance.getConcrete()->getKind() == ProtocolConformanceKind::Normal)
+    specialized = conformance.subst(argType, QuerySubstitutionMap{subs},
```

Comment:
I'm just making this up/guessing, in an attempt to get a specialized conformance.